includes/classes/MyDot/Connector.php

```
@@ -0,0 +1,837 @@
+	 * @param \WP_REST_Request $request The REST request object.
+	 * @return bool True if valid JWT token is present, false otherwise.
+	 */
+	public static function validate_jwt_token_in_request_with_fallback( $request ) {
+		if ( ! $request instanceof \WP_REST_Request ) {
+			return false;
+		}
+
+		// Extract the JWT token from the Authorization header.
+		$auth_header = $request->get_header( 'Authorization' );
+		if ( ! empty( $auth_header ) ) {
+			if ( count( $parts ) === 2 && 'Bearer' === $parts[0] ) {
```

PHPStan (2.1.33)

Undefined variable: $parts


(variable.undefined)